js/trace_test.ts

```diff
@@ -0,0 +1,45 @@
+import { testPerm, assertEqual } from "./test_util.ts";
+import * as deno from "deno";
+
+testPerm({ write: true }, async function traceFunctionSuccess() {
+  const ops = await deno.trace(async () => {
+    const enc = new TextEncoder();
+    const data = enc.encode("Hello");
+    // Mixing sync and async calls
+    const filename = deno.makeTempDirSync() + "/test.txt";
+    await deno.writeFile(filename, data, 0o666);
+    await deno.removeSync(filename);
+  });
+  assertEqual(ops.length, 3);
+  assertEqual(ops[0], "MakeTempDir");
+  assertEqual(ops[1], "WriteFile");
+  assertEqual(ops[2], "Remove");
```

Comment:
Instead of only returning the name of the op, you could return a `OpInfo[]`. And for starters I think it would be nice to know if the op was sync.
```typescript
interface OpInfo {
  sync: bool;
  name: string;
}
```
What do you think?

Reply:
Sounds good (better info for debugging purposes)